Sources/SwiftParser/Statements.swift

```diff
@@ -408,11 +411,14 @@ extension Parser {
   ///
   ///     throw-statement → 'throw' expression
   @_spi(RawSyntax)
-  public mutating func parseThrowStatement(throwHandle: RecoveryConsumptionHandle) -> RawThrowStmtSyntax {
+  public mutating func parseThrowStatement(
+    throwHandle: RecoveryConsumptionHandle,
+    with tryPrefix: (RawUnexpectedNodesSyntax?, RawTokenSyntax)?
```

Comment:
What do you think about naming the elements of the tuple so we don’t have the `.0` and `.1` access below?

Also, I would prefer to name the parameter `tryPrefix` (instead of the external argument name `with` because nothing really suggests what `with` should be).

---

```suggestion
    tryPrefix: (unexpectedBeforeTry: RawUnexpectedNodesSyntax?, tryToken: RawTokenSyntax)?
```

(same for the other functions)

Reply:
I can agree with your argument for the recovery handle but for the `tryPrefix`, I’m still not sure. If the `tryPrefix` took a parameter of type `RawTryTokenSyntax` (which we don’t have), I would be on your side. But as it stands `RawTokenSyntax` is completely ambiguous as to which token it is and if I consider the public signature of this function, nothing indicates that you should pass a the `try` token here. Or, putting it another way: How would the API look like if we also allowed an `await` prefix? You surely don’t want to have two `with` parameters,  right?